esrally/tracker/index.py

```diff
@@ -104,3 +113,20 @@ def extract(client, outdir, index_pattern):
             }
         )
     return results
+
+
+def datastreamextract(client, datastream_pattern):
+    """
+    Calls Elasticsearch client get_data_stream function to retrieve list of indexes
+    :param client: Elasticsearch client
+    :param index_pattern: name of datastream
+    :return: index creation dictionary
+    """
+    results = []
+    # the response might contain multiple indices if a wildcard was provided
+    params_defined = {"expand_wildcards": "all", "filter_path": "data_streams.name"}
+    results_datastreams = client.indices.get_data_stream(name=datastream_pattern, params=params_defined)
+
+    for indices in results_datastreams["data_streams"]:
```

Comment:
This will not return a list of index names. This will return a list of datastream names matching the pattern.

See this:

```python
In [5]: client.indices.get_data_stream('*')
Out[5]:
{'data_streams': [{'name': 'logs-app_search.analytics-default',
   'timestamp_field': {'name': '@timestamp'},
   'indices': [{'index_name': '.ds-logs-app_search.analytics-default-2022.06.16-000001',
     'index_uuid': 'ho8pn4ouS2iRJCwptLumXw'}],
   'generation': 1,
   'status': 'GREEN',
   'template': 'logs-app_search.analytics-default',
   'ilm_policy': 'logs-app_search.analytics-default',
   'hidden': False,
   'system': False,
   'allow_custom_routing': False,
   'replicated': False},
  {'name': 'logs-enterprise_search.api-default',
   'timestamp_field': {'name': '@timestamp'},
   'indices': [{'index_name': '.ds-logs-enterprise_search.api-default-2022.06.16-000001',
     'index_uuid': 'JXbLddJwSHqqKzRVjUS6Tw'},
    {'index_name': '.ds-logs-enterprise_search.api-default-2022.06.23-000002',
     'index_uuid': 'ix8hJeIRT3eIGU1lzD05Cw'}],
   'generation': 2,
   'status': 'GREEN',
   'template': 'logs-enterprise_search.api-default',
   'ilm_policy': 'logs-enterprise_search.api-default',
   'hidden': False,
   'system': False,
   'allow_custom_routing': False,
   'replicated': False}]}
```

Here is what you need to do to get index names:

```python
In [8]: [ix['index_name'] for ds in client.indices.get_data_stream('*')['data_streams'] for ix in ds['indices']]
Out[8]:
['.ds-logs-app_search.analytics-default-2022.06.16-000001',
 '.ds-logs-enterprise_search.api-default-2022.06.16-000001',
 '.ds-logs-enterprise_search.api-default-2022.06.23-000002']
```

Reply:
You are correct, I updated the naming of variables to be more meaningful

Reply:
It is a list of indices indeed. It reads a data-stream list and extracts all indices inside